apps/meteor/server/services/federation/infrastructure/matrix/Bridge.ts

```diff
@@ -235,6 +235,162 @@ export class MatrixBridge implements IFederationBridge {
 		}
 	}
 
+	public async sendThreadMessage(
+		externalRoomId: string,
+		externalSenderId: string,
+		message: IMessage,
+		relatesToEventId: string,
+	): Promise<string> {
+		const text = this.escapeEmojis(
+			await toExternalMessageFormat({
+				message: message.msg,
+				externalRoomId,
+				homeServerDomain: this.internalSettings.getHomeServerDomain(),
+			}),
+		);
+		try {
+			const messageId = await this.bridgeInstance
+				.getIntent(externalSenderId)
+				.matrixClient.sendRawEvent(externalRoomId, MatrixEventType.ROOM_MESSAGE_SENT, {
+					'msgtype': 'm.text',
+					'body': this.escapeEmojis(message.msg),
+					'formatted_body': text,
+					'format': 'org.matrix.custom.html',
+					'm.relates_to': {
+						'rel_type': 'm.thread',
+						'event_id': relatesToEventId,
+						'is_falling_back': true,
+						'm.in_reply_to': {
+							event_id: relatesToEventId,
+						},
+					},
+				});
+			return messageId;
+		} catch (e) {
+			throw new Error('User is not part of the room.');
+		}
```

Comment:
this masks the actual error. in many scenarios can make troubleshooting a nightmare, might even impossible